packages/masked-input/src/index.js

```diff
@@ -0,0 +1,56 @@
+import React from 'react'
+import { Rifm } from 'rifm'
+import { Input } from 'pcln-design-system'
+import creditcards from 'creditcards'
+
+export const MaskedInput = ({ value, onChange, format, ...props }, ref) => (
+  <Rifm
+    value={value}
+    onChange={onChange}
+    format={format}
+    children={({ value, onChange }) => (
+      <Input type="text" {...props} value={value} onChange={onChange} />
+    )}
+  />
+)
+
+MaskedInput.displayName = 'MaskedInput'
+MaskedInput.isField = true
+
+const { card } = creditcards
+
+export const formatCreditCard = str => {
+  const n = str.replace(/[^\d]+/gi, '')
+  return n ? card.format(n, '-') : ''
+}
+
+export const CreditCardInput = props => (
+  <MaskedInput
+    format={formatCreditCard}
+    autoComplete="cc-number"
+    placeholder="4242-4242-4242-4242"
```

Comment:
Is there another alternative to use for this placeholder value instead of 4242-***? (Wonder if 4242 implies any thing to the placeholder value.)

Reply:
We should change this to match whatever design wants

Reply:
agree - I've seen # as a placeholder for numbers a lot of places, I think it is well understood. But this affords Wes and the product team the option to choose a mask, and potentially to A/B test it if they are so inclined.